plugins/plugin-docker/che-plugin-docker-compose/src/main/java/org/eclipse/che/plugin/docker/compose/yaml/ComposeEnvironmentParser.java

```diff
@@ -0,0 +1,173 @@
+        Map<String, CheServiceImpl> services = Maps.newHashMapWithExpectedSize(composeEnvironment.getServices().size());
+        for (Map.Entry<String, ComposeServiceImpl> composeServiceEntry : composeEnvironment.getServices()
+                                                                                           .entrySet()) {
+            ComposeServiceImpl service = composeServiceEntry.getValue();
+
+            CheServiceBuildContextImpl buildContext = null;
+            if (service.getBuild() != null) {
+                buildContext = new CheServiceBuildContextImpl().withContext(service.getBuild().getContext())
+                                                               .withDockerfilePath(service.getBuild().getDockerfile())
+                                                               .withArgs(service.getBuild().getArgs());
+            }
+
+            CheServiceImpl cheService = new CheServiceImpl().withBuild(buildContext)
+                                                            .withCommand(service.getCommand())
+                                                            .withContainerName(service.getContainerName())
+                                                            .withDependsOn(service.getDependsOn())
+                                                            .withEntrypoint(service.getEntrypoint())
+                                                            .withEnvironment(service.getEnvironment())
+                                                            .withExpose(service.getExpose())
+                                                            .withImage(service.getImage())
+                                                            .withLabels(service.getLabels())
+                                                            .withLinks(service.getLinks())
+                                                            .withMemLimit(service.getMemLimit())
+                                                            .withNetworks(service.getNetworks())
+                                                            .withPorts(service.getPorts())
+                                                            .withVolumes(service.getVolumes())
+                                                            .withVolumesFrom(service.getVolumesFrom());
+
+            services.put(composeServiceEntry.getKey(), cheService);
+        }
+        return new CheServicesEnvironmentImpl().withServices(services);
+    }
+
+
+    /**
+     * Checks that object reference is not null, throws {@link IllegalArgumentException} otherwise.
+     *
+     * <p>Exception uses error message built from error message template and error message parameters.
+     */
+    private static void checkNotNull(Object object, String errorMessageTemplate, Object... errorMessageParams) {
+        if (object == null) {
+            throw new IllegalArgumentException(format(errorMessageTemplate, errorMessageParams));
```

Comment:
Wrong exception, see `Preconditions.checkNotNull` | `Objects.requireNonNull`.
You don't need this method, reuse existing one.